sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -177,6 +177,8 @@ def reset_on_fork(restart_thread: true)
             @pid = pid
             spans.clear
             @thread = restart_thread ? Thread.new { work } : nil
+          rescue ThreadError
+            @metrics_reporter.add_to_counter('otel.bsp.failure', labels: { 'reason' => 'ThreadError' })
```

Comment:
Do we also want to log here?

Reply:
I think so, did you have any thoughts on the counter name?  Is `otel.bsp.failure` fine?

Reply:
`otel.bsp.error` might be better